gym_ignition/base/robot/robot_joints.py

```diff
@@ -0,0 +1,361 @@
+
+        Returns:
+            The initial joint positions.
+        """
+
+    @abstractmethod
+    def set_initial_positions(self, positions: np.ndarray) -> bool:
+        """
+        Set the initial joint positions.
+
+        Note that this does not actuate the robot, it only stores the initial joint
+        positions values.
+
+        Args:
+            positions: The initial joint positions.
+
+        Returns:
+             True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def joint_position(self, joint_name: str) -> float:
+        """
+        Return the generalized position of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+
+        Returns:
+            The joint generalized position.
+
+        """
+
+    @abstractmethod
+    def joint_velocity(self, joint_name: str) -> float:
+        """
+        Return the generalized velocity of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+
+        Returns:
+            The joint generalized velocity.
+
+        """
+
+    @abstractmethod
+    def joint_positions(self) -> np.ndarray:
+        """
+        Return the generalized positions of the joints.
+
+        The order matches :meth:`joint_names`.
+
+        Returns:
+            The joint generalized positions.
+        """
+
+    @abstractmethod
+    def joint_velocities(self) -> np.ndarray:
+        """
+        Return the generalized velocities of the joints.
+
+        The order matches :meth:`joint_names`.
+
+        Returns:
+            The joint generalized velocities.
+        """
+
+    @abstractmethod
+    def joint_pid(self, joint_name: str) -> Union[PID, None]:
+        """
+        Return the PID associated to the specified joint.
+
+        Args:
+            joint_name: The joint name.
+
+        Returns:
+            The low-level joint PID or None if no PID was configured.
+        """
+
+    @abstractmethod
+    def dt(self) -> float:
+        """
+        Returns the step_size of the low-level PID controllers.
+
+        Returns:
+            The PIDs step_size.
+        """
+
+    @abstractmethod
+    def set_dt(self, step_size: float) -> bool:
+        """
+        Set the step size used by the low-level PIDs.
+
+        Args:
+            step_size: The PIDs step size.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_position_target(self, joint_name: str, position_reference: float) -> \
+            bool:
+        """
+        Set the position reference of the specified joint.
+
+        Args:
+            joint_name: The name of the joint.
+            position_reference: The joint position reference.
+
+        Returns:
+            True if successful, False otherwise.
+        """
+
+    @abstractmethod
+    def set_joint_velocity_target(self, joint_name: str, velocity_reference: float) -> \
```

Comment:
Naming doubt: why `set_joint_velocity_target` sets a target for a low-level PID, while `set_joint_position_target` sets the reference for a trajectory generator, while for setting a target for a low-level position pid you use `set_joint_position` ?

Reply:
I updated the methods obtaining the following:

- `set_joint_{position,velocity}` implement PID control
- `set_joint_interpolated_position` for the references filtered with the trajectory generator